### ISSUE TITLE
Error handling wrappers with `**kwargs` syntax

### ISSUE THREAD

**Issue by stinodego** [python/mypy#15321]
**Bug Report**

mypy seems to be unable to pass down the types of `**kwargs` arguments properly.

**To Reproduce**

```python
def my_function(flag: bool = True, **kwargs: str) -> None:
    ...


def wrapper(**kwargs: str) -> None:
    my_function(**kwargs)
```

[Playground Link](https://mypy-play.net/?mypy=latest&python=3.11&gist=ac4f704e31a8f4714728e6d229ecf38b)

**Actual Behavior**

```
main.py:6: error: Argument 1 to "my_function" has incompatible type "**Dict[str, str]"; expected "bool"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

**Expected Behavior**

I would expect no errors. Though I can see that inputting `flag="hello"` could cause issues, although the error message does not help in identifying this. Also, in that case, I would expect rewriting the wrapper like below would satisfy mypy (it doesn't):

```python
def wrapper(**kwargs: str) -> None:
    if "flag" in kwargs:
        raise
    my_function(**kwargs)
```

**Your Environment**

- Mypy version used: 1.3.0
- Python version used: 3.11.2

**Comment by erictraut** [comment]
Mypy is correct here. The signature of the `my_function` function indicates that it requires one argument of type `bool`, but you are not providing any such argument when calling it. If you modify your code to supply this `bool` value, mypy no longer emits an error.

```python
def wrapper(**kwargs: str) -> None:
    my_function(True, **kwargs)
```

or

```python
def wrapper(**kwargs: str) -> None:
    my_function(flag=True, **kwargs)
```

**Comment by AlexWaygood** [comment]
Eric is correct

**Comment by stinodego** [comment]
I made a small error in reproducing a minimal working example. If you give `flag` a default value, the same error occurs (I updated the original issue). If you make it a keyword-only argument, the same error occurs.

**Comment by erictraut** [comment]
Yes, the updated sample demonstrates a real bug in mypy. This type checks fine in pyright.

Looks like you can work around the bug by adding a positional-only separator.

```python
def my_function(flag: bool = True, /, **kwargs: str) -> None:
    ...
```

**Comment by stinodego** [comment]
Thanks for verifying! Unfortunately, having the flag keyword-only is not an option for our use case.

**Comment by stinodego** [comment]
@AlexWaygood could this be re-opened?

**Comment by AlexWaygood** [comment]
Not sure I agree that this is definitely a mypy bug, but I don't have time to verify/explain right now, so will reopen for now
